File: files/CH-078.R
library(tidyverse)
library(readxl)

path = "files/CH-078 Extract from Text 2.xlsx"

input = read_xlsx(path, range = "B3", col_names = F) %>%
  pull()
test = read_xlsx(path, range = "B6:B11") %>%
  mutate(`Email Address` = str_sub(`Email Address`, 4))

patterns = c(
  "\\d{4}-\\d{2}-\\d{2}",
  "\\d{2}\\/\\d{2}\\/\\d{4}",
  "\\b\\w+ \\d{1,2}[a-z]*?(?: to \\w+ \\d{1,2}[a-z]*)?, \\d{4}\\b"
)

result = input %>%
  str_extract_all(str_c(patterns, collapse = "|")) %>%
  map(~ .x[.x != ""]) 

result = tibble(`Email Address` = result[[1]])


identical(result, test)
# [1] TRUE

File: files/000-099/078/CH-078.py
import pandas as pd
import re

path = "CH-078 Extract from Text 2.xlsx"

input = pd.read_excel(path, usecols="B", header=None, skiprows=2, nrows=1).iloc[0, 0]

test = pd.read_excel(path, usecols="B", skiprows=5, nrows=6)
test["Email Address"] = test["Email Address"].str[3:]
test = test.sort_values(by="Email Address").reset_index(drop=True)

patterns = [
    r"\d{4}-\d{2}-\d{2}",
    r"\d{2}\/\d{2}\/\d{4}",
    r"\b\w+ \d{1,2}[a-z]*?(?: to \w+ \d{1,2}[a-z]*)?, \d{4}\b"
]

result = pd.DataFrame()
for pattern in patterns:
    result = result.append(pd.DataFrame(re.findall(pattern, input), columns=["Email Address"]))
result = result.sort_values(by="Email Address").reset_index(drop=True)

print(result.equals(test))  # True
